src/einsteinpy/symbolic/tensor.py

```diff
@@ -67,7 +67,7 @@ class Tensor:
     Base Class for Tensor manipulation
     """
 
-    def __init__(self, arr, config="ll"):
+    def __init__(self, arr, config="ll", name=None):
```

Comment:
Hey @Starninja666 I guess you misunderstood the issue?

Reply:
I know what the problem is. The `name` parameter is defined in `BaseRelativityTensor` while `__str__()` is there in `Tensor` :D

We would need to move `name` parameter to `Tensor` or `__str__()` to the derived class, either way it works. But personally, I like the first solution.

Reply:
First solution anyday!

Reply:
No. The motive is to put in the name in `__str__` and not just updating the  name variable!

Reply:
Read this: https://www.journaldev.com/22460/python-str-repr-functions for more information about what is a string representation of a class.

Reply:
I’ll have an extensive look tomorrow about what can be done

Reply:
I'm resolving this.
I suggested the few changes required in a separate review. No need to clutter :)